### PR TITLE
2060_acceptance_review

### PR DESCRIPTION
**Pull request by ijiraq**
Improved documentation. Passes most integration tests.

### CODE REVIEW

File: vos/vos/commands/vcat.py [hunk 1]
```diff
@@ -1,53 +1,61 @@
...
-    usage = "%prog [options] vos:VOSpace/node_name"
-    description = "Writes the content of vos:VOSpace/node_name to stdout."
 
-    parser = CommonParser(usage, description=description)
-    parser.add_option("-q", help="run quietly, exit on error without message", action="store_true")
+def vcat():
 
-    (opt, args) = parser.parse_args()
-    parser.process_informational_options()
+    parser = CommonParser(description=description)
```
Comment (andamian) [inline vos/vos/commands/vcat.py]: description is not defined and it's currently failing.

Isn't it simpler to make the description the doctring of the vcat function and pass it to CommonParser as CommonParser(description=vcat.__doc__)? (double underscore around doc)

File: vos/vos/commands/vcp.py [hunk 2]
```diff
@@ -1,82 +1,86 @@
...
+vcp can be used to cutout particular parts of a FITS file if the VOSpace server supports the action.
+
+extensions and pixel locations accessed with [] brackets:
+vcp vos:Node/filename.fits[3][1:100,1:100] ./
+or
+RA/DEC regions accessed vcp vos:Node/filename.fits(RA, DEC, RAD)
+where RA, DEC and RAD are all given in degrees
+
+If no X509 certificate given on commnad line then default location will be used.
+"""
```
Comment (andamian) [inline vos/vos/commands/vcp.py]: What is the default location?

File: vos/vos/commands/vcp.py [hunk 3]
```diff
@@ -1,82 +1,86 @@
...
+    cutout_pattern = re.compile(r'(.*?)(?P<cutout>(\[[\-+]?[\d*]+(:[\-+]?[\d*]+)?'
+                                r'(,[\-+]?[\d*]+(:[\-+]?[\d*]+)?)?\])+)$')
 
-    ra_dec_cutout_pattern = re.compile("([^\(\)]*?)"
+    ra_dec_cutout_pattern = re.compile("([^()]*?)"
                                        "(?P<cutout>\("
-                                       "(?P<ra>[\-\+]?\d*(\.\d*)?),"
-                                       "(?P<dec>[\-\+]?\d*(\.\d*)?),"
+                                       "(?P<ra>[\-+]?\d*(\.\d*)?),"
+                                       "(?P<dec>[\-+]?\d*(\.\d*)?),"
```
Comment (andamian) [inline vos/vos/commands/vcp.py]: Backslash removed unintentionally? They seem to be needed.

File: vos/vos/commands/vls.py [hunk 4]
```diff
@@ -1,165 +1,164 @@
...
-            try:
-                length = float(size)
-                scale = int(math.log(length) / math.log(1024))
-                length = "%.0f%s" % (length / (1024.0 ** scale), size_unit[scale])
-            except:
-                length = str(int(size))
-        else:
-            length = str(int(size))
-        return "%12s " % length
+def get_terminal_size():
```
Comment (andamian) [inline vos/vos/commands/vls.py]: Is this used anywhere?

File: vos/vos/commonparser.py [hunk 5]
```diff
@@ -1,92 +1,89 @@
...
-            logger = logging.getLogger('vos')
-            logger.setLevel(logging.DEBUG)
+    if sys.version_info[1] > 6:
+        logging.getLogger().addHandler(logging.NullHandler())
 
-        if sys.version_info[1] > 6:
-            logger = logging.getLogger()
-            logger.addHandler(logging.NullHandler())
 
+class CommonParser(argparse.ArgumentParser):
```
Comment (andamian) [inline vos/vos/commonparser.py]: Too bad we could not come to a common approach with the rest of the CADC apps. In essence, this does the same thing (although differently) with the code in cadcutils.

File: vos/vos/commands/vcp.py (hunk 3 above)
Comment (ijiraq) [inline vos/vos/commands/vcp.py]: The backslashes are not needed here.  At least that's what my testing shows.

File: vos/vos/commands/vls.py (hunk 4 above)
Comment (ijiraq) [inline vos/vos/commands/vls.py]: hmm. I think get_terminal_size  was in use at one time.  Perhaps not any more.  The idea was to do column display of file names.  I think I couldn't get this working the way I wanted and so never used in production.

File: vos/vos/commonparser.py (hunk 5 above)
Comment (ijiraq) [inline vos/vos/commonparser.py]: One parser to rule them all!

File: vos/vos/commands/vcp.py (hunk 2 above)
Comment (ijiraq) [inline vos/vos/commands/vcp.py]: Default location should be spelled out as ${HOME}/.ssl/cadcproxy.pem

**Comment by ijiraq** [comment]
I've pushed changes to address the issues you raised and fix the DESCRIPTION variable.  However I still use imports with implicit path structure.  We should discuss this and figure out if there is best practise somewhere.

**Comment by andamian** [comment]
https://stackoverflow.com/questions/4209641/absolute-vs-explicit-relative-import-of-python-module


On Jul 28, 2017, at 11:30 AM, JJ Kavelaars <notifications@github.com<mailto:notifications@github.com>> wrote:


I've pushed changes to address the issues you raised and fix the DESCRIPTION variable. However I still use imports with implicit path structure. We should discuss this and figure out if there is best practise somewhere.

**Pull request merged**